### PR TITLE
Add Node 18 to msal-node supported versions

### LINKED ISSUE #4603: Please provide support for Node JS 17 18 to msal-node
### Core Library

MSAL Node (@azure/msal-node)

### Wrapper Library

Not Applicable

### Description

Please provide Node JS 17 and 18 support for msal-node. My builds are failing because I am using 17.x.x version of node js and msal is 1.7.0

### Source

External (Customer)

### PR DESCRIPTION
**Pull request by WikiRik**
Based on https://github.com/AzureAD/microsoft-authentication-library-for-js/pull/3550

Whitespace changes in `.github/workflows/msal-node-e2e.yml` were automatic, but I can revert those if needed.

### CODE REVIEW

File: change/@azure-msal-node-4f97ea54-b034-4c03-84dc-a4bce5ea639c.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "Add Node 18 to msal-node supported versions",
```
Comment (jasonnutter) [inline change/@azure-msal-node-4f97ea54-b034-4c03-84dc-a4bce5ea639c.json]:
```suggestion
  "comment": "Add Node 18 to msal-node supported versions #4727",
```

**Review by jasonnutter — APPROVED** [review]
This looks great, thanks!

**Comment by sameerag** [comment]
Thank you @WikiRik. You are just on time :), appreciate the contribution.

**Review by sameerag — APPROVED** [review]
lgtm

**Comment by WikiRik** [comment]
Thanks for the approvals! If it's alright with you I will leave the updating (and eventually merging) of the branch up to you so I won't flood this with merge commits or re-request reviews because of rebasing the branch.

**Comment by sameerag** [comment]
> Thanks for the approvals! If it's alright with you I will leave the updating (and eventually merging) of the branch up to you so I won't flood this with merge commits or re-request reviews because of rebasing the branch.

Sure. We will merge it soon and include this with our May release.

**Review by tnorling — APPROVED** [review]

**Pull request merged**